File: src/ZoomLoop.Core/Models/Vehicle.cs
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace ZoomLoop.Core.Models;

public class Vehicle
{
    public Vehicle()
    {
        VIN = string.Empty;
        StockNumber = string.Empty;
        Trim = string.Empty;
        ExteriorColor = string.Empty;
        InteriorColor = string.Empty;
        Transmission = string.Empty;
        FuelType = string.Empty;
        DriveType = string.Empty;
        BodyType = string.Empty;
        Description = string.Empty;
        Images = [];
        Features = [];
    }

    public Guid VehicleId { get; set; }
    public string VIN { get; set; }
    public string StockNumber { get; set; }
    public Guid MakeId { get; set; }
    public Guid VehicleModelId { get; set; }
    public int Year { get; set; }
    public string Trim { get; set; }
    public int Mileage { get; set; }
    public string ExteriorColor { get; set; }
    public string InteriorColor { get; set; }
    public string Transmission { get; set; }
    public string FuelType { get; set; }
    public string DriveType { get; set; }
    public string BodyType { get; set; }
    public int Doors { get; set; }
    public int Seats { get; set; }
    public decimal? EngineSize { get; set; }
    public int? Cylinders { get; set; }
    public int? Horsepower { get; set; }
    public decimal? CityFuelConsumption { get; set; }
    public decimal? HighwayFuelConsumption { get; set; }
    public string Description { get; set; }
    public bool IsNew { get; set; }
    public bool IsCertified { get; set; }
    public DateTime? ManufactureDate { get; set; }
    public Make? Make { get; set; }
    public VehicleModel? VehicleModel { get; set; }
    public List<VehicleImage> Images { get; set; }
    public List<VehicleFeature> Features { get; set; }
}

File: src/ZoomLoop.App/src/app/models/vehicle.ts
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

export interface Vehicle {
  vehicleId?: string;
  vin: string;
  stockNumber: string;
  makeId?: string;
  vehicleModelId?: string;
  year: number;
  trim: string;
  mileage: number;
  exteriorColor: string;
  interiorColor: string;
  transmission: string;
  fuelType: string;
  driveType: string;
  bodyType: string;
  doors: number;
  seats: number;
  engineSize?: number;
  cylinders?: number;
  horsepower?: number;
  cityFuelConsumption?: number;
  highwayFuelConsumption?: number;
  description: string;
  isNew: boolean;
  isCertified: boolean;
  manufactureDate?: Date;
  images: VehicleImage[];
  features: VehicleFeature[];
}

export interface VehicleImage {
  vehicleImageId?: string;
  vehicleId?: string;
  imageUrl: string;
  thumbnailUrl: string;
  caption: string;
  displayOrder: number;
  isPrimary: boolean;
  createdDate: Date;
}

export interface VehicleFeature {
  vehicleFeatureId?: string;
  vehicleId?: string;
  name: string;
  category: string;
  description: string;
  isStandard: boolean;
}

export interface Make {
  makeId?: string;
  name: string;
  logoUrl: string;
  isActive: boolean;
}

export interface VehicleModel {
  vehicleModelId?: string;
  makeId?: string;
  name: string;
  description: string;
  isActive: boolean;
}
